### DISCUSSION TITLE
Browse issue/pull request from URL

### DISCUSSION THREAD

**Discussion by yyoncho** [magit/forge#754] [Ideas]
Thank you for this amazing package!

Sorry if this feature is available, I was unable to find it. What I am looking for is the ability to jump to the `forge` topic from the github url. E. g. when I receive a mail/message containing `https://github.com/foo/bar/issues/1961` I would like to be able to visit directly from that place instead of doing that from `magit-status` buffer.

**Comment by shindere** [comment]
I'd be interested in this, too.

Well, I'd be able to browse in 'forge' issues for repositories I have
not actually cloned.

I am blind and, for me, reading topics and posts, browsing them, is much
more easy in emacs than in my browser.

So, I would love to be able to browse issues even for repositories I am
not following in particular.

Thanks!

**Comment by tarsius** [comment]
A minimal viable version along the lines of what @yyoncho described shouldn't be too hard, but coming up with something that actually dwims instead of refusing to do its job in quite a few situations would be harder.

> Well, I'd be able to browse in 'forge' issues for repositories I have not actually cloned.

So that's a bit of a problem because forge originally was designed with the assumption that you would always clone the repository before dealing with it somehow. That has been lifted to an extend, but you still have to download all the topics not only so that you can list them all but also so that you can look at single one by number.

For "normal sized" repositories it doesn't take that long to download but even then it is still quite noticeable. Given the original request I would have stuck to an implementation that just gives up when the repository hasn't been fetched yet.

> I am blind and, for me, reading topics and posts, browsing them, is much more easy in emacs than in my browser.

I am happy to hear that I could help with that a bit. So I feel a bit bad that I cannot promise implementing any of the features requested here any time soon. As I mentioned in the other issue, I am rather busy, so it is not likely that it will focus on this anytime soon. Sorry.

**Comment by shindere** [comment]
No problem, no need to feel bad! :) It's already very good to know that
you are not opposed to the feature in principle!

So, am I correct that it is already possible to only download the list
of topics for a repository, without actually cloning it? If yes, how can
that be achieved?

**Comment by tarsius** [comment]
> So, am I correct that it is already possible to only download the list of topics for a repository, without actually cloning it? If yes, how can that be achieved?

I have no experience with the workflow but you basically add `M-x forge-add-repository` and provide an URL. Then you can use `' l r` (`forge-list-repositories`), followed by `RET` (`forge-visit-repositories`) to visit the repository/project on the current line. If `forge` is aware of where that is checked out, then it uses `magit-status`, but if not, then it just lists issues belonging to that project. From that buffer you should be able to invoke other `forge` commands.

And I have just update `forge-visit-repository` to read a repository/project when point isn't on something that represents one. So you can now jump directly to a project without having to go through the list of projects.

**Comment by shindere** [comment]
That's awesome Jonas! Thank you so much!

**Comment by tarsius** [comment]
You're welcome!

**Comment by tarsius** [comment]
I've added `forge-visit-topic-from-url`. It's only a minimal viable implementation -- I've now realized that that already goes a long way.

**Comment by shindere** [comment]
A bunch of thanks for this!
